src/hugr/views/sibling.rs

```diff
@@ -0,0 +1,572 @@
+            self.base
+                .node_ports(n, dir)
+                .filter(move |&p| self.is_boundary_port(n, p))
+                .map(move |p| (n, p))
+        })
+    }
+
+    /// The signature of the subgraph.
+    pub fn signature(&self) -> AbstractSignature
+    where
+        Base: HugrView,
+    {
+        let (input, input_static): (Vec<_>, Vec<_>) = self
+            .incoming_ports()
+            .filter_map(|(n, p)| self.base.get_optype(n).signature().get(p))
+            .partition_map(|edgekind| match edgekind {
+                EdgeKind::Value(t) => Either::Left(t),
+                EdgeKind::Static(t) => Either::Right(t),
+                _ => unimplemented!("ControlFlow and StateOrder edges not supported"),
+            });
+        let output = self
+            .outgoing_ports()
+            .filter_map(|(n, p)| self.base.get_optype(n).signature().get(p))
+            .map(|edgekind| {
+                if let EdgeKind::Value(t) = edgekind {
+                    t
+                } else {
+                    unimplemented!(
+                        "ControlFlow, StateOrder and Static edges not supported for output"
+                    )
+                }
+            })
+            .collect_vec();
+        AbstractSignature::new(input, output, input_static)
+    }
+
+    /// The parent of the sibling subgraph.
+    pub fn get_parent(&self) -> Node
+    where
+        Base: HugrView,
+    {
+        self.base
+            .get_parent(self.nodes[0])
+            .expect("invalid subgraph")
+    }
+
+    /// Construct a [`SimpleReplacement`] to replace `self` with `replacement`.
+    ///
+    /// `replacement` must be a hugr with DFG root and its signature must
+    /// match the signature of the subgraph.
+    ///
+    /// We currently do not support inputs of the replacement graph being
+    /// copied.
+    ///
+    /// May return one of the following five errors
+    ///  - [`InvalidReplacement::InvalidDataflowGraph`]: the replacement
+    ///    graph is not a [`crate::ops::OpTag::DataflowParent`]-rooted graph,
+    ///  - [`InvalidReplacement::InvalidDataflowParent`]: the replacement does
+    ///    not have an input and output node,
+    ///  - [`InvalidReplacement::InvalidSignature`]: the signature of the
+    ///    replacement DFG does not match the subgraph signature,
+    ///  - [`InvalidReplacement::NonConvexSubgrah`]: the sibling subgraph is not
+    ///    convex, or
+    ///  - [`InvalidReplacement::InvalidCopy`]: the replacement has a copy at
+    ///    the input boundary.
+    pub fn create_simple_replacement(
+        &self,
+        replacement: Hugr,
+    ) -> Result<SimpleReplacement, InvalidReplacement>
+    where
+        Base: HugrView,
+    {
+        let removal = self.nodes().iter().copied().collect();
+
+        let rep_root = replacement.root();
+        let dfg_optype = replacement.get_optype(rep_root);
+        if dfg_optype.tag() != OpTag::Dfg {
```

Comment:
```suggestion
        if !OpTag::Dfg.is_superset(dfg_optype.tag())  {
```
It's currently the same, but for future-proofing.